lib/logstash/inputs/neo4j.rb

```diff
@@ -54,6 +54,13 @@ def run(queue)
     end
   end # def run
 
+  def stop
+    super
+    if @schedule
```

Comment:
It would be more defensive if this check was for `@scheduler` instead of `@schedule`. This probably won't cause any problems now, but I see potential for a race down the road.